layers/+tools/lsp/funcs.el

```diff
@@ -46,3 +46,101 @@ https://github.com/emacs-lsp/lsp-javascript/issues/9#issuecomment-379515379"
       (all-completions (company-grab-symbol) candidates)))
   (make-local-variable 'company-transformers)
   (add-to-list 'company-transformers 'lsp-prefix-company-transformer))
+
+(defun spacemacs/lsp-bind-keys-for-mode (mode)
+  "Define key bindings for the specific MODE."
+  (spacemacs/declare-prefix-for-mode mode "m=" "format")
+  (spacemacs/declare-prefix-for-mode mode "me" "execute")
+  (spacemacs/declare-prefix-for-mode mode "mg" "goto")
+  (spacemacs/declare-prefix-for-mode mode "mh" "hierarchy")
+  (spacemacs/declare-prefix-for-mode mode "mj" "jump")
+  (spacemacs/declare-prefix-for-mode mode "ml" "lsp/backend")
+  (spacemacs/declare-prefix-for-mode mode "mr" "refactor")
+  (spacemacs/declare-prefix-for-mode mode "mT" "toggle")
+
+  (spacemacs/set-leader-keys-for-major-mode mode
+    ;;format
+    "=b" #'lsp-format-buffer
+    ;;execute
+    "ea" #'lsp-execute-code-action
+    ;;goto
+    "gi" #'lsp-goto-implementation
+    "gt" #'lsp-goto-type-definition
+    ;;hierarchy
+    "hh" #'lsp-describe-thing-at-point
+    ;;jump
+    "ja" #'spacemacs/lsp-avy-document-symbol
```

Comment:
What is the difference between jump and goto? I feel like all the jump keybindings should go under goto.

Reply:
Yeah you're right. At some point I felt goto was in danger of getting overcrowded and made an artificial distinction between actions with only one possible target (goto) and those with multiple potential targets (jump - lsp-ui-peek). Moved all the `j` bindings back under `g` (had to rebind lsp-imenu under `m g m` as `m g i` was already bound to goto implementation).